gateway/1.4/modules/ROOT/pages/flex-conn-jenkins-cli.adoc

```diff
@@ -0,0 +1,342 @@
+
+== Before You Begin
+Before you can deploy a Jenkins Pipeline, complete the following tasks:
+
+. xref:flex-install.adoc[Install Flex Gateway].
+
+. xref:access-management::connected-apps-developers.adoc#create-a-connected-app[Configure a Connected App]:
+
+.. Use the *App acts on its behalf (client credentials)* type and include the following scopes:
+
+*** API Manager: *Manage APIs Configuration*, *Manage Policies*, *View Policies*, and *Deploy API Proxies*
+*** Runtime Manager: *Read Servers* and *Manage Servers*
+*** Exchange: *Exchange Viewer*
+*** General: *View Organization*
+
+.. Save the *Id* and *Secret* of the Connected app you configure.
+
+. Collect the group ID, asset ID, and asset version (GAV) of the Exchange asset you want to create or apply.
++
+To collect your GAV from Exchange:
++
+.. Go to Exchange.
+.. Find the asset to add or apply.
+.. Collect the group ID and asset ID from the URL.
++
+For example, the IDs collected from the API Manager API Exchange URL, `+https://anypoint.mulesoft.com/exchange/portals/anypoint-platform/f1e97bc6-315a-4490-82a7-23abe036327a.anypoint-platform/api-manager-api/minor/1.0/pages/home/+`, are:
++
+*** Group ID: `f1e97bc6-315a-4490-82a7-23abe036327`
+*** Asset ID: `api-manager-api`
+
+.. Collect the asset version from *Latest Stable*.
++
+For example, the version collected from the API Gateway HTTP basic authentication policy template is `1.3.1`.
+
+. Collect the following information from your Anypoint Platform instance:
+
+** *Organization ID* for the organization where you want to run Flex Gateway
++
+For more information, see xref:access-management::organization.adoc#find-your-organization-id[Find Your Organization ID].
+
+** *Environment ID* for the environment where you want to run Flex Gateway
++
+For more information, see xref:api-manager::latest-overview-concept.adoc#what-api-manager-looks-like[What API Manager Looks Like].
+
+
+== Initial Configuration
+
+Before you can create and deploy an API instance to your Flex Gateway, you must complete the following steps:
+
+. <<define-useful-functions, Define Useful Functions>>.
+. <<define-useful-parameters, Define Useful Parameters>>.
+. <<install-anypoint-cli-4-x, Install Anypoint CLI 4.x>>
+. <<register-and-run, Register and Run>>.
+
+After you implement these Pipeline stages, you can customize your Pipeline by making API Manager CLI requests to manage APIs.
+
+=== Define Useful Functions
+
+To begin, define these useful functions in your Jenkinsfile:
+
+[source,ssh]
+----
+def parseYamlFile(String file) {
+   object = readYaml file: file
+   return object
+}
+ 
+def getAPIInstanceID(String file) {
+   return readFile(file: file).split(' ').last().replace('\n', '').trim()
+}
+----
+
+=== Define Useful Parameters
+
+Since the commands in the Jenkins Pipeline reuses some parameters, it is helpful to define them at the beginning of your Jenkinsfile.
```

Comment:
```suggestion
Since the commands in the Jenkins Pipeline reuse some parameters, it is helpful to define them at the beginning of your Jenkinsfile.
```